controller/api/destination/watcher/profile_watcher.go

```diff
@@ -127,53 +127,76 @@ func (pw *ProfileWatcher) deleteProfile(obj interface{}) {
 		Name:      profile.Name,
 	}
 
-	publisher, ok := pw.getProfilePublisher(id)
+	publisher, ok := pw.getProfilePublisher(&id)
 	if ok {
 		publisher.update(nil)
 	}
 }
 
-func (pw *ProfileWatcher) getOrNewProfilePublisher(id ProfileID, profile *sp.ServiceProfile) *profilePublisher {
-	pw.Lock()
-	defer pw.Unlock()
+func (pw *ProfileWatcher) getOrNewProfilePublisher(id *ProfileID, profile *sp.ServiceProfile) *profilePublisher {
```

Comment:
it looks like all the changes in this file are due to switching ProfileID to a pointer and properly handling nil.  Why did this need to change?  As far as I can tell we shouldn't ever be passing `nil` into subscribe.  (all of the instances in server.go pass in a reference to a specific struct).

Reply:
These changes can be removed. I simplified this earlier today, and with that simplification we don't need to pass `nil` profiles anymore. Thanks for catching that!